### PR TITLE
Simplified parallel structure

### PR DESCRIPTION
**Pull request by garrettmflynn**
This PR improves the parallel demo by removing the extra `threading` to accommodate the Flask _and_ HTTP relay server. Since the Flask server is already an HTTP server, we don't need both of them. We can simply use Flask to relay events, then allow that to block the main thread until complete. 

This more closely mirrors how we'd handle things on the GUIDE.

### CODE REVIEW

File: src/tqdm_publisher/_demos/_parallel_bars/_server.py [hunk 1]
```diff
@@ -214,50 +192,41 @@ def listen_to_events():
 app = Flask(__name__)
 cors = CORS(app)
 app.config["CORS_HEADERS"] = "Content-Type"
-PORT = find_free_port()
+PORT = 3768  # find_free_port()
```
Comment (CodyCBakerPhD) [inline src/tqdm_publisher/_demos/_parallel_bars/_server.py]: Do you prefer this to be hardcoded? If so I'd remove the comment

Comment (CodyCBakerPhD) [reply src/tqdm_publisher/_demos/_parallel_bars/_server.py]: And would we even need the `find_free_port` logic anymore or can that be cleaned up too?

**Comment by CodyCBakerPhD** [comment]
Looks good, code is much cleaner, just one minor comment about the port stuff

File: src/tqdm_publisher/_demos/_parallel_bars/_server.py (hunk 1 above)
Comment (garrettmflynn) [reply src/tqdm_publisher/_demos/_parallel_bars/_server.py]: Good point, this isn't even a function in the codebase anymore. Removed!

**Comment by CodyCBakerPhD** [comment]
@garrettmflynn Some conflicts for some odd reason

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**